hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -4000,13 +4020,46 @@ else if ( !supportsRowValueConstructorSyntaxInInList() ) {
 			}
 		}
 		else {
+			int nullExpressions = 0;
```

Comment:
How about we do all of this only when the LHS has a model part type of `java.lang.Class`?